File: Scripts/factory_method/ConcreteCreator.php
<?php
require_once('Creator.php');
require_once('Circle.php');
require_once('Square.php');
require_once('Triangle.php');

class ConcreteCreator implements Creator {

	//Factory method implementation
	public function create($condition){
		$product = null;
		
		if($condition == "circle"){
			$product = new Circle();
		}else if($condition == "square"){
			$product = new Square();
		}else if($condition == "triangle"){
			$product = new Triangle();
		}
		
		return $product;
	}
}
?>

File: src/factory_method/ConcreteCreator.java
package factory_method;

public class ConcreteCreator implements Creator {
    @Override
    public Product create(String condition) {
        Product product;
        switch (condition){
            case "Triangle":product =  new Triangle();break;
            case "Circle":product =  new Circle();break;
            case "Square":product =  new Square();break;
            default: product=null;
        }
        return product;
    }
}
